networkx/algorithms/coloring/greedy_coloring.py

```diff
@@ -262,6 +276,11 @@ def greedy_color(G, strategy=strategy_largest_first, interchange=False):
     """
     colors = {}  # dictionary to keep track of the colors of the nodes
 
+    if nx.utils.is_string_like(strategy):
```

Comment:
Hmm how about

    if strategy in GREEDY_COLOR_STRATEGIES:
        strategy = GREEDY_COLOR_STRATEGIES[strategy]
    if not callable(strategy):
        raise NetworkXError('strategy must be one of the built-in strings or a function')  # or something

This covers the case where the string doesn't appear in the dictionary, and where the strategy keyword argument is more generally a non-callable thing.